src/js/structs/ServiceTree.js

```diff
@@ -120,6 +121,21 @@ module.exports = class ServiceTree extends Tree {
         });
       }
 
+      if (filter.labels != null && filter.labels.length > 0) {
+        services = services.filter(function (service) {
+          return filter.labels.some(function (label) {
+            let serviceLabels = ServiceUtil.convertServiceLabelsToArray(
+              service
+            );
+
+            return serviceLabels.find(function (serviceLabel) {
```

Comment:
This `.find` could be replaced by `.some` to get rid of the ` != null` at the end.